src/direct-binary-upload-options.js

```diff
@@ -241,11 +241,13 @@ class DirectBinaryUploadOptions {
      * Retrieves the path to the folder where the files will be uploaded. This value
      * is based on the URL that was provided to the options.
      *
+     * The path value will not be URL encoded.
+     *
      * @returns {string} Full path to a folder on the target.
      */
     getTargetFolderPath() {
         const { pathname } = URL.parse(this.getUrl());
-        return pathname;
+        return decodeURIComponent(pathname);
```

Comment:
Main thing to look for is double decoding. I think this looks ok.

Reply:
@bobvanmanen The unit tests include tests involving the `+` character (which can alternatively be interpreted as a space, and is a prime candidate for double decoding). I was originally using `decodeURI()`, but that method doesn't decode `%2B` into a `+`; so I changed it to `decodeURIComponent()`, which does.

I think we should be safe. I did a bunch of manual testing around `+` and several other characters that have been problematic - everything checked out.